pkg/sdn/plugin/subnets.go

```diff
@@ -95,11 +95,6 @@ func (master *OsdnMaster) deleteNode(nodeName string) error {
 	if err != nil {
 		return fmt.Errorf("Error fetching subnet for node %q for deletion: %v", nodeName, err)
 	}
-	_, ipnet, err := net.ParseCIDR(sub.Subnet)
-	if err != nil {
-		return fmt.Errorf("Error parsing subnet %q for node %q for deletion: %v", sub.Subnet, nodeName, err)
-	}
-	master.subnetAllocator.ReleaseNetwork(ipnet)
```

Comment:
May be separate assign/revoke subnet from addNode/removeNode methods, then you can
- call revoke subnet from watchSubnets and
- call revoke subnet + delete node object from watchNodes